app/renderer/components/Inspector/Actions.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import _ from 'lodash';
+import { Row, Col, Button, Select, Modal, Input, notification, } from 'antd';
+import { actionDefinitions, actionArgTypes } from './shared';
+
+const { STRING, NUMBER } = actionArgTypes;
+
+const Option = { Select };
+
+export default class Actions extends Component {
+
+  startPerformingAction (actionName, action) {
+    const { startEnteringActionArgs, applyClientMethod } = this.props;
+    if (_.isEmpty(action.args)) {
+      applyClientMethod({methodName: action.methodName, args: [], skipScreenshotAndSource: !action.refresh});
+    } else {
+      startEnteringActionArgs(actionName, action);
+    }
+  }
+
+  executeCommand () {
+    const { pendingAction, cancelPendingAction, applyClientMethod, t } = this.props;
+    let {args, action} = pendingAction;
+    let {methodName} = action;
+
+    // Special case for 'startActivity'
+    // TODO: Fix these... args aren't getting through
+    if (action.methodName === 'startActivity') {
+      args = {appPackage: args[0], appActivity: args[1], appWaitPackage: args[2],
+              intentAction: args[3], intentCategory: args[4], intentFlags: args[5],
+              optionalIntentArguments: args[6], dontStopAppOnReset: args[7]};
+    }
+
+    // Special case for 'rotateDevice'
+    if (action.methodName === 'rotateDevice') {
+      args = {x: args[0], y: args[1], duration: args[2], radius: args[3], rotation: args[4], touchCount: args[5]};
+    }
+
+    // Special case for 'execute'
+    if (action.methodName === 'execute') {
+      if (!_.isEmpty(args[1])) {
+        try {
+          args[1] = JSON.parse(args[1]);
+        } catch (e) {
+          notification.error({
+            message: `${t('Invalid JSON')}: args[1]}`,
+            duration: 5,
+          });
+        }
+      }
+    }
+
+    applyClientMethod({methodName, args, skipScreenshotAndSource: !action.refresh});
+    cancelPendingAction();
+  }
+
+  render () {
+    const {
+      t, selectActionGroup, selectSubActionGroup, selectedActionGroup, selectedSubActionGroup, pendingAction,
+      cancelPendingAction, setActionArg } = this.props;
+    return [
+      <Row gutter={16} style={{marginBottom: '8px'}}>
+        <Col span={24}>
+          <Select style={{width: '100%'}} onChange={(actionGroupName) => selectActionGroup(actionGroupName)} placeholder="Select Action Group">
```

Comment:
why width is not in css?

Reply:
placehodler text should be translated

Reply:
I'll make a note to move all the style tags to css rules for tomorrow.